tests/test_schema.py

```diff
@@ -654,8 +654,8 @@ def test_only_in_init(SchemaClass, user):
     assert 'age' in s
 
 def test_invalid_only_param(user):
```

Comment:
The below demonstrates an actual change in behavior.

However, I think this is valid. If we do `only=('foo',)` and `foo` is actually a declared field, then we do not get any sort of exception. For example, doing:

```python
UserSchema(only=('name',)).dump({})
```

would not cause an exception.

Thus, it seems odd that exceptions only get raised when the field is not declared on the schema.

Reply:
With https://github.com/marshmallow-code/marshmallow/pull/810/commits/443aa4ea00ad3183c71011f87ef5497e91df4736, we now throw at schema instantiation time if a field in `only` or `exclude` is not present on the schema class.